app/notification_email.go

```diff
@@ -160,17 +159,15 @@ func getGroupMessageNotificationEmailSubject(user *model.User, post *model.Post,
 			"Day":         t.Day,
 			"Year":        t.Year,
 		}
-		subjectText = translateFunc("app.notification.subject.group_message.full", subjectParameters)
-	} else {
-		var subjectParameters = map[string]interface{}{
-			"SiteName": siteName,
-			"Month":    t.Month,
-			"Day":      t.Day,
-			"Year":     t.Year,
-		}
-		subjectText = translateFunc("app.notification.subject.group_message.generic", subjectParameters)
+		return translateFunc("app.notification.subject.group_message.full", subjectParameters)
 	}
-	return subjectText
+	var subjectParameters = map[string]interface{}{
+		"SiteName": siteName,
+		"Month":    t.Month,
+		"Day":      t.Day,
+		"Year":     t.Year,
+	}
+	return translateFunc("app.notification.subject.group_message.generic", subjectParameters)
```

Comment:
Looks good to me other than for those similar maps that are being created every time.

Like `subjectParameters` on `getGroupMessageNotificationEmailSubject`, would it be idiomatic to do it like:
```go
func getGroupMessageNotificationEmailSubject(user *model.User, post *model.Post, translateFunc i18n.TranslateFunc, siteName string, channelName string, emailNotificationContentsType string, useMilitaryTime bool) string {
	t := getFormattedPostTime(user, post, useMilitaryTime, translateFunc)
	subjectParameters := map[string]interface{}{
		"SiteName": siteName,
		"Month":    t.Month,
		"Day":      t.Day,
		"Year":     t.Year,
	}

	if emailNotificationContentsType == model.EMAIL_NOTIFICATION_CONTENTS_FULL {
		subjectParameters["ChannelName"] = channelName
		return translateFunc("app.notification.subject.group_message.full", subjectParameters)
	}

	return translateFunc("app.notification.subject.group_message.generic", subjectParameters)
}
```

Same case on `getNotificationEmailBody` where identical maps are being created.  Maybe have a map declared on top of those using it and then reuse or add member as necessary.
```go
info := map[string]interface{}{
    "Hour":     t.Hour,
    "Minute":   t.Minute,
    "TimeZone": t.TimeZone,
    "Month":    t.Month,
    "Day":      t.Day,
}

// add field if necessary
info["SenderName"] = senderName
```

Reply:
Thanks for the feedback @saturninoabril. I updated the code.